### PR TITLE
Showing matched rule on the page

### PR DESCRIPTION
**Pull request by hrmtsh2**
Additionally fix an error in dealing with optional fields that are of the type 'array'

### CODE REVIEW

File: src/components/RequestInput.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <script setup>
-import { ref } from 'vue';
+import { ref, toRaw } from 'vue';
```
Comment (oliverdunk) [inline src/components/RequestInput.vue]: Could you explain why we need `toRaw` here? Does the proxied object not work with JSON.stringify?

File: src/stores/rulesStore.js [hunk 2]
```diff
@@ -30,7 +30,8 @@ export const useRulesStore = defineStore('rules', {
     parsedRulesList: [],
     rulesetFilesUploaded: false,
     requestMatched: false,
-    matchedRule: {},
+    matchedRuleString: '',
+    matchedRuleStringSet: false,
```
Comment (oliverdunk) [inline src/stores/rulesStore.js]: Hmm, thoughts on just representing this with `undefined` as the value of `matchedRuleString`? It seems odd to have an entirely separate bool.

File: src/stores/rulesStore.js [hunk 3]
```diff
@@ -200,7 +217,7 @@ export const useRulesStore = defineStore('rules', {
           matchedRules.push(this.parsedRulesList[i]);
         }
       }
-      this.requestMatched = true;
+      // this.requestMatched = true;
```
Comment (oliverdunk) [inline src/stores/rulesStore.js]: Do we still need this?

**Review by oliverdunk — CHANGES_REQUESTED** [review]
Thanks for opening this! Left an initial review.

**Comment by hrmtsh2** [comment]
I have made the requested changes, kindly have a look!

File: src/stores/rulesStore.js (hunk 3 above)
Comment (oliverdunk) [reply src/stores/rulesStore.js]: @hrmtsh2, I think this is still there?

Comment (hrmtsh2) [reply src/stores/rulesStore.js]: My bad, have fixed it

File: src/components/MatchedRuleBox.vue [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  white-space: pre-wrap;
+  display: flex;
+}
+
+pre {
+  width: 100%;
+  height: 100%;
+  margin: 0;
+  font-family: monospace;
+  color: white;
```
Comment (oliverdunk) [inline src/components/MatchedRuleBox.vue:40]: In light theme, this makes the text invisible. Could we either choose a color that works in both or have some styles based on [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme)?

File: src/components/RequestInput.vue [hunk 5]
```diff
@@ -8,22 +8,31 @@ const httpMethod = ref('GET');
...
+const matchedRule = ref(null);
+let matchedRuleString = '';
 
 function submitRequest(ev) {
   ev.preventDefault();
+
+  // Headers validity checking
+  let requestHeaders = {};
+  try {
+    requestHeaders = JSON.parse(headers.value);
```
Comment (oliverdunk) [inline src/components/RequestInput.vue]: I think we need some handling here for if the headers field is empty?

**Review by oliverdunk — CHANGES_REQUESTED** [review]
Thanks for the fixes! Left a few more comments.

**Comment by hrmtsh2** [comment]
Could you verify whether the modification for light mode works now?

File: src/components/MatchedRuleBox.vue (hunk 4 above)
Comment (hrmtsh2) [reply src/components/MatchedRuleBox.vue:40]: Could you verify whether the modification for light mode works now?

File: src/components/RequestInput.vue [hunk 6]
```diff
@@ -27,11 +27,16 @@ function submitRequest(ev) {
...
+    headers: requestHeaders,
     body: body.value
   };
   matchedRule.value = rulesStore.requestMatcher(formObject)[0];
-  matchedRuleString = JSON.stringify(matchedRule.value.rule, null, 2);
+  // When matchedRule.value is undefined, it means no matching rule was found
+  try {
+    matchedRuleString = JSON.stringify(matchedRule.value.rule, null, 2);
+  } catch (e) {
+    window.alert('No matching rule found.');
```
Comment (oliverdunk) [inline src/components/RequestInput.vue:38]: If we don't return here, but we didn't update the variable, doesn't that mean we will update `matchedRuleString` to be the previously matched rule?

File: src/components/MatchedRuleBox.vue (hunk 4 above)
Comment (oliverdunk) [reply src/components/MatchedRuleBox.vue:40]: Looking much better!

**Review by oliverdunk — APPROVED** [review]
Left a nit, but I don't see it as a blocker, so I'll merge. Thanks!

**Pull request merged**